components/automate-cli/cmd/chef-automate/reindexing.go

```diff
@@ -0,0 +1,627 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+type Indices []struct {
+	Health       string `json:"health"`
+	Status       string `json:"status"`
+	Index        string `json:"index"`
+	UUID         string `json:"uuid"`
+	Pri          string `json:"pri"`
+	Rep          string `json:"rep"`
+	DocsCount    string `json:"docs.count"`
+	DocsDeleted  string `json:"docs.deleted"`
+	StoreSize    string `json:"store.size"`
+	PriStoreSize string `json:"pri.store.size"`
+}
+
+type IndexSettingsVersion struct {
+	Settings struct {
+		Index struct {
+			Version struct {
+				CreatedString  string `json:"created_string"`
+				UpgradedString string `json:"upgraded_string"`
+			} `json:"version"`
+		} `json:"index"`
+	} `json:"settings"`
+}
+
+// Define the main structure to hold the dynamic index names
+type IndexSettings map[string]interface{}
+
+var infoReindexing = `
+Reindexing of Elasticsearch/OpenSearch indices if needed.
+`
+
+func init() {
+	reindexCmd.SetUsageTemplate(infoReindexing)
+	RootCmd.AddCommand(reindexCmd)
+}
+
+var reindexCmd = &cobra.Command{
+	Use:   "reindex",
+	Short: "Reindex Elasticsearch indices if needed",
+	RunE:  runReindex,
+}
+
+var isReindexing bool
+var mu sync.Mutex
+
+func runReindex(cmd *cobra.Command, args []string) error {
+	fmt.Println("Reindexing Elasticsearch/OpenSearch indices.")
+	mu.Lock()
+	if isReindexing {
+		mu.Unlock()
+		fmt.Println("Reindexing is already in progress. Please wait for it to complete.")
+		return nil
+	}
+	isReindexing = true
+	mu.Unlock()
+
+	defer func() {
+		mu.Lock()
+		isReindexing = false
+		mu.Unlock()
+	}()
+
+	indices, err := fetchIndices()
+	if err != nil {
+		return err
+	}
+
+	for _, index := range indices {
+		// // Apply your filtering logic here
+		if strings.HasPrefix(index.Index, "security-auditlog") || strings.HasPrefix(index.Index, ".opendistro") {
```

Comment:
We need to add these two as well. .plugins-ml-config, .opensearch-observability. They are getting creaed as part of os 2.18